py4j-python/src/py4j/java_gateway.py

```diff
@@ -221,7 +221,8 @@ def launch_gateway(port=0, jarpath="", classpath="", javaopts=[],
                    die_on_exit=False, redirect_stdout=None,
                    redirect_stderr=None, daemonize_redirect=True,
                    java_path="java", create_new_process_group=False,
-                   enable_auth=False, cwd=None, return_proc=False):
+                   enable_auth=False, cwd=None, return_proc=False,
+                   gateway_class="py4j.GatewayServer"):
```

Comment:
this should be called entry_point_class. Strictly speaking, we are specifying the Application's Entry Point